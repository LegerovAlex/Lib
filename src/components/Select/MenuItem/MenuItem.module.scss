.menuItem {
  display: flex;
  align-items: center;
  padding: 6px 16px;
  font-size: 1rem;
  line-height: 1.5;
  color: #000000de;
  cursor: pointer;
  min-height: 36px;
  user-select: none;
  transition: all 0.3s ease;

  &:hover {
    background-color: #f5f5f5;
  }
}

.disabled {
  color: #aaa;
  cursor: not-allowed;
  pointer-events: none;
}
