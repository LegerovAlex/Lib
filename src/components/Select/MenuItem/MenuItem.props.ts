import { ReactNode } from 'react';

export interface MenuItemProps {
  value: string;
  children: ReactNode;
  disabled: boolean;
  onSelect: (value: string) => void;
}
