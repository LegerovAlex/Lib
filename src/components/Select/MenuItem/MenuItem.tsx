import { FC } from 'react';
import { MenuItemProps } from './MenuItem.props';
import styles from './MenuItem.module.scss';

export const MenuItem: FC<MenuItemProps> = ({ value, onSelect, children, disabled }) => {
  const handleClick = () => {
    if (disabled) return;
    onSelect(value);
  };
  return (
    <li className={`${styles.menuItem} ${disabled && styles.disabled}`} onClick={handleClick}>
      {children}
    </li>
  );
};
