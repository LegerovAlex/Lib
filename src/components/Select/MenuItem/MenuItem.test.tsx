import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MenuItemProps } from './MenuItem.props';
import { MenuItem } from './MenuItem';

export interface RenderMenuItemResult {
  menuItem: HTMLElement;
  onSelect: jest.Mock;
}

export function renderMenuItem(props: Partial<MenuItemProps> = {}): RenderMenuItemResult {
  const onSelect = jest.fn();

  render(
    <MenuItem
      {...props}
      value={props.value ?? 'test'}
      onSelect={onSelect}
      disabled={props.disabled}
    >
      {props.children ?? 'test'}
    </MenuItem>,
  );

  const menuItem = screen.getByRole('listitem');

  return { menuItem, onSelect };
}

describe('MenuItem', () => {
  it('renders with correct text', () => {
    const { menuItem } = renderMenuItem({ children: 'Children' });
    expect(menuItem).toBeInTheDocument();
    expect(menuItem).toHaveTextContent('Children');
  });
  it('calls onSelect with corret value on click', async () => {
    const { menuItem, onSelect } = renderMenuItem({ children: 'Children', value: 'value' });
    await userEvent.click(menuItem);
    expect(onSelect).toHaveBeenCalledTimes(1);
    expect(onSelect).toHaveBeenCalledWith('value');
  });
  it('does not call onSelect when disabled', async () => {
    const { menuItem, onSelect } = renderMenuItem({
      children: 'Children',
      value: 'value',
      disabled: true,
    });
    await userEvent.click(menuItem);
    expect(onSelect).not.toHaveBeenCalled();
    expect(onSelect).not.toHaveBeenCalledWith('value');
  });
  it('matches snapshot when enabled', () => {
    const { menuItem } = renderMenuItem({ children: 'Snapshot Test', value: 'snap' });
    expect(menuItem).toMatchSnapshot();
  });
  it('matches snapshot when disabled', () => {
    const { menuItem } = renderMenuItem({
      children: 'Snapshot Disabled',
      value: 'snap',
      disabled: true,
    });
    expect(menuItem).toMatchSnapshot();
  });
});
