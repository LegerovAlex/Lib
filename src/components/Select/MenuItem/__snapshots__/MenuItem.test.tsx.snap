// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MenuItem matches snapshot when disabled 1`] = `
<li
  class="menuItem disabled"
>
  Snapshot Disabled
</li>
`;

exports[`MenuItem matches snapshot when enabled 1`] = `
<li
  class="menuItem undefined"
>
  Snapshot Test
</li>
`;
